### DISCUSSION TITLE
Outputting dynamic components from CMS

### DISCUSSION THREAD

**Discussion by AAA55555** [nuxt/nuxt#24634] [Questions (Nuxt 2)]
I receive a text string from Directus CMS and inside the text is the name of the component <call-to-action></call-to-action>. Previously, I had it done via v-html, but now there is a component and it does not render it. Added config for Wabpack to nuxt.config.js/build - NUXT 2.17.1

```
extend(config) {
    config.resolve.alias['vue$'] = 'vue/dist/vue.esm.js'
  
```
  
and then brought it out through
  ```
<script>
import { normalizeContent } from '~/src/utils'

export default {
  props: {
    content: {
      type: String,
      required: true,
    },
  },
  computed: {
    dynamicComponent() {
      return {
        template: `<div><client-only>${normalizeContent(
          this.content
        )}</client-only></div>`,
      }
    },
  },
  render(createElement) {
    return createElement('div', [createElement(this.dynamicComponent)])
  },
}
</script>
```
But I read in the documentation that the bundle is increasing by 10kb, how else can I solve my problem?

**Comment by TheAlexLichter** [comment]
I don't think you need the runtime compiler here.

Instead, you could use `<component :is="dynamicComponent">`, *but*, you'd have to make all the components globally available (e.g. putting them in `~/components/global`

**Reply by AAA55555** [reply]
fecit  ```<component :is="test" />``` , and added it to computed
```
computed: {
    date() {
      return normalizeDate(this.page?.date)
    },
    test() {
      return '<call-to-action-cms>123</call-to-action-cms>'
    },
  },
  ```
and it shows me
  ```
  <article> 
  <call-to-action-cms>
  123
  </call-to-action-cms>
  <!--<call-to-action-cms-->
  123
  </article>
  ```
for some reason it displays duplication, I don’t know why, at least according to the code it should be displayed once

**Reply by TheAlexLichter** [reply]
`is` should only contain the component name, e.g. `test` should be `call-to-action-cms`
Then you can render the content via slot `<component :is="test">123</component>`

**Reply by AAA55555** [reply]
I have just such an option, text and inside the component name ``` text text text <call-to-action-cms></call-to-action-cms> ```
